src/Search/Microsoft.Azure.Search/Customizations/Documents/DocumentsOperations.cs

```diff
@@ -899,12 +940,11 @@ internal DocumentsOperations(SearchIndexClient client)
 
             // Construct URL
             bool useGet = Client.UseHttpGetForQueries;
-            var baseUrl = this.Client.BaseUri.AbsoluteUri;
-            var url =
-                new Uri(
-                    new Uri(baseUrl + (baseUrl.EndsWith("/") ? "" : "/")),
-                    (useGet ? "docs" : "docs/search.post.search")).ToString();
-
+            var baseUrl = Client.BaseUri;
+            var url = baseUrl + (baseUrl.EndsWith("/") ? "" : "/") + (useGet ? "docs" : "docs/search.post.search");
+            url = url.Replace("{searchServiceName}", Client.SearchServiceName);
+            url = url.Replace("{searchDnsSuffix}", Client.SearchDnsSuffix);
+            url = url.Replace("{indexName}", Client.IndexName);
```

Comment:
I'd rather not since this code needs to mirror the generated code as closely as possible. This makes it easier to apply changes to this code that newer versions of AutoRest apply to the generated code.